### PR TITLE
Parser:Remove CONSTRAINT syntax for KEY | INDEX | FULLTEXT

### PR DESCRIPTION
**Pull request by wty4427300**
### What problem does this PR solve? 

#535 

### Check List 

Tests 

 - No code

Code changes

 - Has exported variable/fields change

### CODE REVIEW

**Comment by wty4427300** [comment]
sql_yacc.yy

table_constraint_def:
          key_or_index opt_index_name_and_type '(' key_list_with_expression ')'
          opt_index_options
          {
            $$= NEW_PTN PT_inline_index_definition(KEYTYPE_MULTIPLE,
                                                   $2.name, $2.type, $4, $6);
          }
        | FULLTEXT_SYM opt_key_or_index opt_ident '(' key_list_with_expression ')'
          opt_fulltext_index_options
          {
            $$= NEW_PTN PT_inline_index_definition(KEYTYPE_FULLTEXT, $3, NULL,
                                                   $5, $7);
          }
            $$= NEW_PTN PT_inline_index_definition(KEYTYPE_SPATIAL, $3, NULL, $5, $7);
          }
        | opt_constraint_name constraint_key_type opt_index_name_and_type
          '(' key_list_with_expression ')' opt_index_options
          {
            /*
              Constraint-implementing indexes are named by the constraint type
              by default.
            */
            LEX_STRING name= $3.name.str != NULL ? $3.name : $1;
            $$= NEW_PTN PT_inline_index_definition($2, name, $3.type, $5, $7);
          }
        | opt_constraint_name FOREIGN KEY_SYM opt_ident '(' key_list ')' references
          {
            $$= NEW_PTN PT_foreign_key_definition($1, $4, $6, $8.table_name,
                                                  $8.reference_list,
                                                  $8.fk_match_option,
                                                  $8.fk_update_opt,
                                                  $8.fk_delete_opt);
          }
        | opt_constraint_name check_constraint opt_constraint_enforcement
          {
            $$= NEW_PTN PT_check_constraint($1, $2, $3);
            if ($$ == nullptr) MYSQL_YYABORT; // OOM
          }
        ;

**Comment by wty4427300** [comment]
PTAL

**Comment by tangenta** [comment]
The unit test is failed. @wty4427300 

BTW, you can use `git push your-remote-repo your-branch -f` to force-push your commits to the repository, so you don't have to open so many PRs.

File: parser.y [hunk 1]
```diff
@@ -8349,6 +8318,38 @@ Constraint:
 		}
 		$$ = cst
 	}
+|	"FULLTEXT" KeyOrIndexOpt IndexName '(' IndexColNameList ')' IndexOptionList
+        	{
```
Comment (zz-jason) [inline parser.y]: please use tab instead of spaces to indent the code.

**Comment by tangenta** [comment]
@wty4427300 Please make sure all tests are passed.

**Pull request closed without merging**